substratest/factory.py

```diff
@@ -346,21 +357,24 @@ class _BaseComputePlanSpec(_Spec, abc.ABC):
     aggregatetuples: typing.List[ComputePlanAggregatetupleSpec]
     testtuples: typing.List[ComputePlanTesttupleSpec]
 
-    def add_traintuple(self, algo, dataset, data_samples, in_models=None, tag=''):
+    def add_traintuple(self, algo, dataset, data_samples, in_models=None, tag='', metadata=None):
         in_models = in_models or []
+        metadata = metadata or {}
```

Comment:
This line does the same as L370, you'll need to use one or the other!

Reply:
This also applies to add_aggregatuple and add_composite_traintuple.

Reply:
I suggest you use everywhere the pattern you're already using for add_testtuple.

Reply:
You're right, I forgot to remove it :/